templates/OperationQueryParamFromClientPropertySnippet.stencil

```diff
@@ -0,0 +1,8 @@
+{% for param in op.params.query.optional %}
+{% if param.implementation == "client" %}
+if let {{ param.key }} = client.{{ param.key }} {
+    {% include "OperationOptionsDecodingSnippet.stencil" %}
+    queryParams.append("{{ param.key }}", {{ param.value}})
+}
+{% endif %}
```

Comment:
```suggestion
{% for param in op.params.query.optional %}
{% if param.implementation == "client" %}
if let {{ param.key }} = {{ param.path }} {
    {% include "OperationOptionsDecodingSnippet.stencil" %}
    queryParams.append("{{ param.key }}", {{ param.value}})
}
{% endif %}
```

Then you would not need to duplicate this code. Internally, you would set the path property as `self.path = "\(implementation.rawValue).{key}"`

Reply:
For some reason, {enum}.rawValue does not work in stencil. I guess rawValue is computed values. 
I have changed to adapt with your suggestion in new iteration.